src/creature.cpp

```diff
@@ -1017,19 +1009,53 @@ bool Creature::setFollowCreature(Creature* creature)
 			onWalkAborted();
 		}
 
-		hasFollowPath = false;
-		forceUpdateFollowPath = false;
 		followCreature = creature;
-		isUpdatingPath = true;
+		creature:addFollowedByCreature(this);
```

Comment:
```suggestion
		creature->addFollowedByCreature(this);
```

Reply:
I was correct then, the changing of : to -> is what the fix was, that is great to know! I still think a parameter name change might also be apt, but this works

Reply:
can u also check why does monster take addictiona randomStep when hasfollowpath || followcreature, for example when changing floors?
![double dancestep](https://github.com/otland/forgottenserver/assets/96269029/b3e89d09-7406-47ad-84e6-3fbc8c9c81be)

Reply:
I am have already fixed that, but have not pushed to this repo yet. I am still working on one thing which is turning out to be a nuisance. Once I fix it I will push all the updates.

Reply:
Also monsters in real Tibia moves instantly when they see player. In TFS they wait event interval.
Same when they die, in TFS they die one by one, in real Tibia all together in the same time. As a result you get weird looking experience points (like many "5exp" for killing rats)
![image](https://github.com/otland/forgottenserver/assets/96269029/cec20142-a71c-4ab8-bf19-797d13420f92)
Do you have any idea how to fix it?

Reply:
Its already fixed wait for the next push